File: frontend/admin/packages/modules/system/src/api/system/admin/v1/base_config.ts
import service from "@liujitcn/kratos-admin-core/request";
import {
  type BaseConfigForm,
  type BaseConfigService,
  type CreateBaseConfigRequest,
  type DeleteBaseConfigRequest,
  type GetBaseConfigRequest,
  type PageBaseConfigRequest,
  type PageBaseConfigResponse,
  type RefreshBaseConfigCacheRequest,
  type SetBaseConfigStatusRequest,
  type UpdateBaseConfigRequest
} from "@liujitcn/kratos-admin-system/rpc/system/admin/v1/base_config";
import type { Empty } from "@liujitcn/kratos-admin-system/rpc/google/protobuf/empty";

const BASE_CONFIG_URL = "/v1/admin/base/config";

/** Admin系统配置服务 */
export class BaseConfigServiceImpl implements BaseConfigService {
  /** 刷新缓存 */
  RefreshBaseConfigCache(request: RefreshBaseConfigCacheRequest): Promise<Empty> {
    return service<RefreshBaseConfigCacheRequest, Empty>({
      url: `${BASE_CONFIG_URL}/cache`,
      method: "put",
      data: request
    });
  }

  /** 查询系统配置分页列表 */
  PageBaseConfig(request: PageBaseConfigRequest): Promise<PageBaseConfigResponse> {
    return service<PageBaseConfigRequest, PageBaseConfigResponse>({
      url: `${BASE_CONFIG_URL}`,
      method: "get",
      params: request
    });
  }

  /** 查询系统配置 */
  GetBaseConfig(request: GetBaseConfigRequest): Promise<BaseConfigForm> {
    return service<GetBaseConfigRequest, BaseConfigForm>({
      url: `${BASE_CONFIG_URL}/${request.id}`,
      method: "get"
    });
  }

  /** 创建系统配置 */
  CreateBaseConfig(request: CreateBaseConfigRequest): Promise<Empty> {
    return service<BaseConfigForm | undefined, Empty>({
      url: `${BASE_CONFIG_URL}`,
      method: "post",
      data: request.base_config
    });
  }

  /** 更新系统配置 */
  UpdateBaseConfig(request: UpdateBaseConfigRequest): Promise<Empty> {
    return service<BaseConfigForm | undefined, Empty>({
      url: `${BASE_CONFIG_URL}/${request.base_config?.id ?? ""}`,
      method: "put",
      data: request.base_config
    });
  }

  /** 删除系统配置 */
  DeleteBaseConfig(request: DeleteBaseConfigRequest): Promise<Empty> {
    return service<DeleteBaseConfigRequest, Empty>({
      url: `${BASE_CONFIG_URL}/${request.id}`,
      method: "delete"
    });
  }

  /** 设置状态 */
  SetBaseConfigStatus(request: SetBaseConfigStatusRequest): Promise<Empty> {
    return service<SetBaseConfigStatusRequest, Empty>({
      url: `${BASE_CONFIG_URL}/${request.id}/status`,
      method: "put",
      data: request
    });
  }
}

export const defBaseConfigService = new BaseConfigServiceImpl();

File: backend/internal/service/system/admin/v1/base_config.go
package admin

import (
	"context"
	"fmt"

	adminv1 "github.com/liujitcn/kratos-admin/backend/api/gen/go/system/admin/v1"
	biz "github.com/liujitcn/kratos-admin/backend/internal/biz/system/admin"
	"github.com/liujitcn/kratos-core/errorsx"

	"github.com/go-kratos/kratos/v3/log"
	"google.golang.org/protobuf/types/known/emptypb"
)

// BaseConfigService Admin系统配置服务
type BaseConfigService struct {
	adminv1.UnimplementedBaseConfigServiceServer
	baseConfigCase *biz.BaseConfigCase
}

// NewBaseConfigService 创建Admin系统配置服务
func NewBaseConfigService(
	configCase *biz.BaseConfigCase,
) *BaseConfigService {
	return &BaseConfigService{
		baseConfigCase: configCase,
	}
}

// RefreshBaseConfigCache 刷新缓存
func (s *BaseConfigService) RefreshBaseConfigCache(ctx context.Context, req *adminv1.RefreshBaseConfigCacheRequest) (*emptypb.Empty, error) {
	err := s.baseConfigCase.RefreshBaseConfig(ctx)
	if err != nil {
		log.Error(fmt.Sprintf("RefreshBaseConfig %v", err))
		return nil, errorsx.WrapInternal(err, "刷新缓存失败")
	}
	return new(emptypb.Empty), nil
}

// PageBaseConfig 查询系统配置分页列表
func (s *BaseConfigService) PageBaseConfig(ctx context.Context, req *adminv1.PageBaseConfigRequest) (*adminv1.PageBaseConfigResponse, error) {
	page, err := s.baseConfigCase.PageBaseConfig(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("PageBaseConfig %v", err))
		return nil, errorsx.WrapInternal(err, "查询系统配置分页列表失败")
	}

	return page, nil
}

// GetBaseConfig 查询系统配置
func (s *BaseConfigService) GetBaseConfig(ctx context.Context, req *adminv1.GetBaseConfigRequest) (*adminv1.BaseConfigForm, error) {
	config, err := s.baseConfigCase.GetBaseConfig(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("GetBaseConfig %v", err))
		return nil, errorsx.WrapInternal(err, "查询系统配置失败")
	}
	return config, nil
}

// CreateBaseConfig 创建系统配置
func (s *BaseConfigService) CreateBaseConfig(ctx context.Context, req *adminv1.CreateBaseConfigRequest) (*emptypb.Empty, error) {
	err := s.baseConfigCase.CreateBaseConfig(ctx, req.GetBaseConfig())
	if err != nil {
		log.Error(fmt.Sprintf("CreateBaseConfig %v", err))
		return nil, errorsx.WrapInternal(err, "创建系统配置失败")
	}
	return new(emptypb.Empty), nil
}

// UpdateBaseConfig 更新系统配置
func (s *BaseConfigService) UpdateBaseConfig(ctx context.Context, req *adminv1.UpdateBaseConfigRequest) (*emptypb.Empty, error) {
	err := s.baseConfigCase.UpdateBaseConfig(ctx, req.GetBaseConfig())
	if err != nil {
		log.Error(fmt.Sprintf("UpdateBaseConfig %v", err))
		return nil, errorsx.WrapInternal(err, "更新系统配置失败")
	}
	return new(emptypb.Empty), nil
}

// DeleteBaseConfig 删除系统配置
func (s *BaseConfigService) DeleteBaseConfig(ctx context.Context, req *adminv1.DeleteBaseConfigRequest) (*emptypb.Empty, error) {
	err := s.baseConfigCase.DeleteBaseConfig(ctx, req.GetId())
	if err != nil {
		log.Error(fmt.Sprintf("DeleteBaseConfig %v", err))
		return nil, errorsx.WrapInternal(err, "删除系统配置失败")
	}
	return new(emptypb.Empty), nil
}

// SetBaseConfigStatus 设置状态
func (s *BaseConfigService) SetBaseConfigStatus(ctx context.Context, req *adminv1.SetBaseConfigStatusRequest) (*emptypb.Empty, error) {
	err := s.baseConfigCase.SetBaseConfigStatus(ctx, req)
	if err != nil {
		log.Error(fmt.Sprintf("SetBaseConfigStatus %v", err))
		return nil, errorsx.WrapInternal(err, "设置状态失败")
	}
	return new(emptypb.Empty), nil
}
